components/camel-service/src/main/java/org/apache/camel/component/service/ServiceEndpoint.java

```diff
@@ -0,0 +1,115 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Consumer;
+import org.apache.camel.DelegateEndpoint;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.api.management.ManagedAttribute;
+import org.apache.camel.api.management.ManagedResource;
+import org.apache.camel.cloud.DiscoverableService;
+import org.apache.camel.cloud.ServiceDefinition;
+import org.apache.camel.cloud.ServiceRegistry;
+import org.apache.camel.cluster.CamelClusterView;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.impl.cloud.DefaultServiceDefinition;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriPath;
+
+/**
+ * Represents an endpoint which only becomes active when the {@link CamelClusterView}
+ * has the leadership.
+ */
+@ManagedResource(description = "Managed Service Endpoint")
+@UriEndpoint(
+    firstVersion = "2.22.0",
+    scheme = "service",
+    syntax = "service:serviceName:delegateUri",
+    consumerClass = ServiceConsumer.class,
+    consumerOnly = true,
+    title = "Service",
+    lenientProperties = true,
+    label = "cloud")
+public class ServiceEndpoint extends DefaultEndpoint implements DelegateEndpoint {
+    private final Endpoint delegateEndpoint;
+    private final ServiceRegistry serviceRegistry;
+    private final Map<String, String> serviceParameters;
+    private final ServiceDefinition serviceDefinition;
+
+    @UriPath(description = "The endpoint uri to expose as service")
+    @Metadata(required = "true")
+    private final String delegateUri;
+
+    public ServiceEndpoint(String uri, ServiceComponent component, ServiceRegistry serviceRegistry, Map<String, String> serviceParameters, String delegateUri) {
+        super(uri, component);
+
+        this.serviceRegistry = serviceRegistry;
+        this.serviceParameters = serviceParameters;
+        this.delegateUri = delegateUri;
+        this.delegateEndpoint = getCamelContext().getEndpoint(delegateUri);
+
+        // The service properties set on uri override parameter provided by a
+        // an endpoint of type DiscoverableService.
+        this.serviceDefinition = computeServiceDefinition(component.getCamelContext(), delegateEndpoint);
+    }
+
+    @ManagedAttribute(description = "The consumer endpoint to expose as a service", mask = true)
+    @Override
+    public Endpoint getEndpoint() {
```

Comment:
I dont think complex objects over JMX is good, eg Endpoint. Should be the string uri of the endpoint returned, eg String type